### PR TITLE
First part of mtls PR

### PR DESCRIPTION
**Pull request by slonka**
The whole PR will introduce mTLS support for compatible services. This is the first part, mainy building blocks, that will enable consecutive parts to execute logic and tests.

### CODE REVIEW

File: envoy-control-tests/src/main/resources/envoy/envoy2_auth_config.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ node:
     use_remote_address: true
     access_log_enabled: false
     add_upstream_external_address_header: true
+    has_static_secrets_defined: true
```
Comment (lukidzi) [inline envoy-control-tests/src/main/resources/envoy/envoy2_auth_config.yaml:26]: I can't find usage of this property.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt [hunk 2]
```diff
@@ -37,11 +43,17 @@ object RandomConfigFile :
 abstract class EnvoyControlTestConfiguration : BaseEnvoyTest() {
```
Comment (pzmi) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt:43]: This class contains setup, http clients, stub applications interactions, SUT factories, SUT interactions, some assertions, and cleanup. I think we should consider splitting it :)

File: envoy-control-tests/src/main/resources/envoy/envoy2_auth_config.yaml (hunk 1 above)
Comment (slonka) [reply envoy-control-tests/src/main/resources/envoy/envoy2_auth_config.yaml:26]: Good catch! I've added too much to this part of the pr, will remove it and add it when it's used.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt (hunk 2 above)
Comment (slonka) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt:43]: I've created an issue for it. I'll work on it once mtls PR is one, ok? :) https://github.com/allegro/envoy-control/issues/113

**Pull request merged**